src/binaryen-c.cpp

```diff
@@ -1842,8 +1843,8 @@ void BinaryenSetMemory(BinaryenModuleRef module, BinaryenIndex initial, Binaryen
   }
 
   auto* wasm = (Module*)module;
-  wasm->memory.initial = initial;
-  wasm->memory.max = maximum;
+  wasm->memory.initial = initial = std::max(initial, numSegments);
```

Comment:
I also don't understand the correction here - numSegments is the number of segments, not the size of memory (also it's not scaled by the page size). As I said above I think we can just drop these corrections.